common/src/main/java/com/wynntils/core/config/ConfigManager.java

```diff
@@ -50,9 +52,16 @@ public final class ConfigManager extends Manager {
     private File userConfig;
     private JsonObject configObject;
 
-    public ConfigManager() {
-        super(List.of());
+    public ConfigManager(ConfigUpfixerManager upfixer) {
+        super(List.of(upfixer));
+
+        // First, we load the config file
         loadConfigFile();
+
+        // Now, we have to apply upfixers, before any config loading happens
+        if (upfixer.runUpfixers(configObject)) {
```

Comment:
make the upfixer run lazily by storing a version in the config and comparing differences.